src/foremast/s3/s3deploy.py

```diff
@@ -107,12 +107,55 @@ def _upload_artifacts_to_version(self):
         """Recursively upload directory contents to S3."""
         if not os.listdir(self.artifact_path) or not self.artifact_path:
             raise S3ArtifactNotFound
-        cmd = 'aws s3 sync {} {} --delete --exact-timestamps --profile {}'.format(self.artifact_path,
-                                                                                  self.s3_version_uri, self.env)
-        result = subprocess.run(cmd, check=True, shell=True, stdout=subprocess.PIPE)
-        LOG.debug("Upload Command Ouput: %s", result.stdout)
+
+        uploaded = False
+        if self.s3props.get("content_encodings"):
+            LOG.info("Uploading in multiple parts to set content encoding")
+            uploaded = self.content_encoding_uploads()
+
+        if not uploaded:
+            cmd = 'aws s3 sync {} {} --delete --exact-timestamps --profile {}'.format(self.artifact_path,
+                                                                                      self.s3_version_uri, self.env)
+            result = subprocess.run(cmd, check=True, shell=True, stdout=subprocess.PIPE)
+            LOG.debug("Upload Command Ouput: %s", result.stdout)
+
         LOG.info("Uploaded artifacts to %s bucket", self.bucket)
 
+    def content_encoding_uploads(self):
```

Comment:
Same as others `content_metadata_uploads`. If we ever need to patch or add more support for other headers we can do it.